src/WebSdk/Publish/Tasks/Tasks/MsDeploy/CreateMSDeployScript.cs

```diff
@@ -19,18 +19,8 @@ public class CreateMSDeployScript : Task
 
         public override bool Execute()
         {
-            if (!File.Exists(ScriptFullPath))
-            {
-                File.Create(ScriptFullPath);
```

Comment:
I don't think you can remove these File.Creates because the directory containing the file might not exist. If it doesn't, File.WriteAllTexts throws an exception.

Reply:
Oh.

I was looking at https://learn.microsoft.com/en-us/dotnet/api/system.io.file.create?view=net-8.0 and that did not suggest that File.Create creates parent directories.

Reply:
I didn't phrase that very well, apologies. I was thinking that this looks like a possible correctness bug in that File.WriteAllText throws an exception when the directory doesn't exist. You're right that File.Create doesn't create it either; I normally use Directory.CreateDirectory for this sort of thing, and I'm assuming that's what the original author was trying to do.